crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -1125,6 +1125,62 @@ def f(fn: Callable[[int], int]) -> None: ...
 f(a)
 ```
 
+### Class literals
+
+#### Classes with metaclasses
+
+```py
+from typing import Callable, overload
+from typing_extensions import Self
+from knot_extensions import TypeOf, static_assert, is_subtype_of
+
+class MetaWithReturn(type):
+    def __call__(cls) -> "A":
+        return super().__call__()
+
+class A(metaclass=MetaWithReturn): ...
+
+# TODO: This should not error
+# This errors because the metaclass `__call__` is not seen as overriding `type.__call__`
+# and we don't yet check `__new__` or `__init__`.
```

Comment:
I agree that this should not error, but for a different reason -- I think (as discussed in Discord) that the text in the spec about ignoring certain metaclass `__call__` methods for purposes of converting a constructor to a callable is making unwarranted assumptions that can lead to false positive errors. I think it is more correct to always respect the signature of the metaclass `__call__` method.

If this causes problems in practice (including with the spec conformance suite, and we can't effectuate an update to the spec), we can consider a future change. But for now I think we should unconditionally use metaclass `__call__` if present (which I think will make this test pass as written and remove the need for the TODO comment?)